src/network.rs

```diff
@@ -78,40 +78,26 @@ fn test_parse_addrtype() {
 }
 
 pub fn parse_unicast_addr(addrtype: &SdpAddrType, value: &str) -> Result<IpAddr, SdpParserError> {
```

Comment:
The addrtype gets parsed before. If the parsing fails the parse_addr_type() function should fail and parse_unicast_addr() should never get called. addrtype is only used in parse_unicast_addr to differentiate which of the two IP address parsers to call. We can get rid of the by simply calling IpAddr::from_str() instead. And then handle the potential error from that in errors.rs.